### PR TITLE
Bug fix for CF and MGF of discretenonparametric

### PR DESCRIPTION
**Pull request by trappmartin**

### CODE REVIEW

**Comment by devmotion** [comment]
Thanks for the PR! I wonder why https://github.com/JuliaStats/Distributions.jl/blob/7b3890d6547385ca460c209b993481335a629995/src/univariate/discrete/discretenonparametric.jl#L50 is not sufficient? In any case, could you add some tests that are broken currently and fixed by the PR?

**Comment by trappmartin** [comment]
Good question, no idea.

**Comment by trappmartin** [comment]
>  In any case, could you add some tests that are broken currently and fixed by the PR?
Sure, can give it a try.

**Comment by trappmartin** [comment]
I added a test to the categorical distribution as I encountered the issue when using it.

File: test/univariate/discrete/categorical.jl [hunk 1]
```diff
@@ -57,6 +57,12 @@ for p in Any[
     @test pdf.(d, support(d)) == p
     @test pdf.(d, 1:k) == p
 
+    @test cf(d, 0) ≈ 1.0
+    @test cf(d, 1) ≈ p' * Distributions.cis.(1:length(p))
```
Comment (devmotion) [inline test/univariate/discrete/categorical.jl:61]: IIRC `cis` is in Base and exported:

```suggestion
    @test cf(d, 1) ≈ p' * cis.(1:length(p))
```

Possibly a bit nicer (also below) would be to use `dot`:

```suggestion
    @test cf(d, 1) ≈ dot(p, cis.(1:length(p))
```
However, possibly this requires to add a `using LinearAlgebra`.

**Review by devmotion — APPROVED** [review]
Thank you!

**Pull request merged**